### PR TITLE
HBASE-26871 shaded mapreduce and shaded byo-hadoop client artifacts contain no classes

### PR DESCRIPTION
**Pull request by busbey**
- revert poms for both modules
- add a test to the shaded artifact validation that checks for java classes

Change is only needed on branch-2 and branch-2.5.

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 55s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +0 :ok: |  shelldocs  |   0m  1s |  Shelldocs was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ branch-2 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 18s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   3m  4s |  branch-2 passed  |
| +1 :green_heart: |  compile  |   0m 38s |  branch-2 passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 11s |  Maven dependency ordering for patch  |
| -1 :x: |  mvninstall  |   2m 19s |  root in the patch failed.  |
| +1 :green_heart: |  compile  |   0m 39s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 39s |  the patch passed  |
| +1 :green_heart: |  shellcheck  |   0m  1s |  There were no new shellcheck issues.  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  xml  |   0m  4s |  The patch has no ill-formed XML file.  |
| -1 :x: |  hadoopcheck  |   4m 15s |  The patch causes 13 errors with Hadoop v3.1.2.  |
| -1 :x: |  hadoopcheck  |   8m 40s |  The patch causes 13 errors with Hadoop v3.2.1.  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 31s |  The patch does not generate ASF License warnings.  |
|  |   |  18m 45s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4279/1/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/4279 |
| Optional Tests | dupname asflicense shellcheck shelldocs javac hadoopcheck xml compile |
| uname | Linux 94af736e2edf 5.4.0-1043-aws #45~18.04.1-Ubuntu SMP Fri Apr 9 23:32:25 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | branch-2 / c73ef2c0c8 |
| Default Java | AdoptOpenJDK-1.8.0_282-b08 |
| mvninstall | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4279/1/artifact/yetus-general-check/output/patch-mvninstall-root.txt |
| hadoopcheck | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4279/1/artifact/yetus-general-check/output/patch-javac-3.1.2.txt |
| hadoopcheck | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4279/1/artifact/yetus-general-check/output/patch-javac-3.2.1.txt |
| Max. process+thread count | 46 (vs. ulimit of 12500) |
| modules | C: hbase-shaded/hbase-shaded-client-byo-hadoop hbase-shaded/hbase-shaded-mapreduce hbase-shaded/hbase-shaded-check-invariants hbase-shaded/hbase-shaded-with-hadoop-check-invariants U: hbase-shaded |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4279/1/console |
| versions | git=2.17.1 maven=3.6.3 shellcheck=0.4.6 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 14s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  6s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ branch-2 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 16s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   3m 23s |  branch-2 passed  |
| +1 :green_heart: |  compile  |   0m 44s |  branch-2 passed  |
| +1 :green_heart: |  shadedjars  |   0m 30s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 41s |  branch-2 passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 15s |  Maven dependency ordering for patch  |
| -1 :x: |  mvninstall  |   3m 14s |  root in the patch failed.  |
| +1 :green_heart: |  compile  |   0m 52s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 52s |  the patch passed  |
| -1 :x: |  shadedjars  |   5m 18s |  patch has 13 errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 39s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 14s |  hbase-shaded-client-byo-hadoop in the patch passed.  |
| +1 :green_heart: |  unit  |   0m 15s |  hbase-shaded-mapreduce in the patch passed.  |
| +1 :green_heart: |  unit  |   0m 10s |  hbase-shaded-check-invariants in the patch passed.  |
| +1 :green_heart: |  unit  |   0m 10s |  hbase-shaded-with-hadoop-check-invariants in the patch passed.  |
|  |   |  19m 12s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4279/1/artifact/yetus-jdk8-hadoop2-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/4279 |
| Optional Tests | unit javac javadoc shadedjars compile |
| uname | Linux 028d2e859862 5.4.0-90-generic #101-Ubuntu SMP Fri Oct 15 20:00:55 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | branch-2 / c73ef2c0c8 |
| Default Java | AdoptOpenJDK-1.8.0_282-b08 |
| mvninstall | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4279/1/artifact/yetus-jdk8-hadoop2-check/output/patch-mvninstall-root.txt |
| shadedjars | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4279/1/artifact/yetus-jdk8-hadoop2-check/output/patch-shadedjars.txt |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4279/1/testReport/ |
| Max. process+thread count | 47 (vs. ulimit of 12500) |
| modules | C: hbase-shaded/hbase-shaded-client-byo-hadoop hbase-shaded/hbase-shaded-mapreduce hbase-shaded/hbase-shaded-check-invariants hbase-shaded/hbase-shaded-with-hadoop-check-invariants U: hbase-shaded |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4279/1/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 20s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  6s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ branch-2 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 17s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   4m  1s |  branch-2 passed  |
| +1 :green_heart: |  compile  |   0m 48s |  branch-2 passed  |
| +1 :green_heart: |  shadedjars  |   0m 31s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 46s |  branch-2 passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 20s |  Maven dependency ordering for patch  |
| -1 :x: |  mvninstall  |   3m 53s |  root in the patch failed.  |
| +1 :green_heart: |  compile  |   0m 57s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 57s |  the patch passed  |
| -1 :x: |  shadedjars  |   5m 39s |  patch has 13 errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 43s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 16s |  hbase-shaded-client-byo-hadoop in the patch passed.  |
| +1 :green_heart: |  unit  |   0m 19s |  hbase-shaded-mapreduce in the patch passed.  |
| +1 :green_heart: |  unit  |   0m 10s |  hbase-shaded-check-invariants in the patch passed.  |
| +1 :green_heart: |  unit  |   0m 10s |  hbase-shaded-with-hadoop-check-invariants in the patch passed.  |
|  |   |  21m 29s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4279/1/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/4279 |
| Optional Tests | unit javac javadoc shadedjars compile |
| uname | Linux 205071fd0632 5.4.0-90-generic #101-Ubuntu SMP Fri Oct 15 20:00:55 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | branch-2 / c73ef2c0c8 |
| Default Java | AdoptOpenJDK-11.0.10+9 |
| mvninstall | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4279/1/artifact/yetus-jdk11-hadoop3-check/output/patch-mvninstall-root.txt |
| shadedjars | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4279/1/artifact/yetus-jdk11-hadoop3-check/output/patch-shadedjars.txt |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4279/1/testReport/ |
| Max. process+thread count | 59 (vs. ulimit of 12500) |
| modules | C: hbase-shaded/hbase-shaded-client-byo-hadoop hbase-shaded/hbase-shaded-mapreduce hbase-shaded/hbase-shaded-check-invariants hbase-shaded/hbase-shaded-with-hadoop-check-invariants U: hbase-shaded |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-4279/1/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

File: hbase-shaded/hbase-shaded-check-invariants/src/test/resources/ensure-jars-have-correct-contents.sh [hunk 1]
```diff
@@ -114,6 +114,10 @@ declare -i bad_artifacts=0
 declare -a bad_contents
 for artifact in "${artifact_list[@]}"; do
   bad_contents=($(jar tf "${artifact}" | grep -v -E "${allowed_expr}" || true))
+  class_count=$(jar tf "${artifact}" | grep -c -E '\.class$' || true)
+  if [ ${#bad_contents[@]} -eq 0 ] && [ "${class_count}" -lt 1 ]; then
+    bad_contents=("The artifact contains no java class files.")
```
Comment (apurtell) [inline hbase-shaded/hbase-shaded-check-invariants/src/test/resources/ensure-jars-have-correct-contents.sh:119]: Good, thanks

**Review by apurtell — APPROVED** [review]

**Pull request merged**